docs/DeveloperGuide.md

```diff
@@ -1,34 +1,178 @@
 # Developer Guide
 
-## Design & implementation
+![Supported Java versions](https://img.shields.io/badge/Java-11-blue.svg)
+![Supported OS](https://img.shields.io/badge/Supported%20OS-Windows|MacOS|Linux-yellow.svg)
+![Java CI](https://github.com/AY2021S1-CS2113T-T09-4/tp/workflows/Java%20CI/badge.svg)
+
+## Table of Contents
+
+- [Introduction](#introduction)
+    * [To be added](#to-be-added)
+        + [To be added](#to-be-added)
```

Comment:
> Is it better to include the links to each section right after it is created so that it won't be missed out in the future?

I agree with `kewenlok`.
Maybe it is better to do so as the table of contents may grow larger than expected?